### ISSUE TITLE
False positive: Sentinel value as default parameter causes type error

### ISSUE THREAD

**Issue by CNSeniorious000** [DetachHead/basedpyright#1142]
### Description

- [pyright playground](https://pyright-play.net/?pythonVersion=3.13&strict=true&code=PoVQcgygogKgBAXjgewEYCsCmBjALgCgEoBYAKDIBNMAzOa5ZfVAQwCcAuOASwDtdE4oSLEKcAdBLhwAxHFwBPAA6ZOXAOY9krTGTL1GhIA)
- [basedpyright playground](https://basedpyright.com/?typeCheckingMode=standard&code=PoVQcgygogKgBAXjgewEYCsCmBjALgCgEoAoYgE0wDM5Lll9UBDAJwC44BLAO10TlEixC7AHRi4cAMRxcATwAOmdhwDmXZM0yla9QkA)

```py
_UNSET = object()

def foo(bar: int = _UNSET): ...  # type: ignore

foo()
^^^^^

Argument of type "object" cannot be assigned to parameter "bar" of type "int" in function "foo"
  "object" is not assignable to "int"  (reportArgumentType)
```

`_UNSET` is being used as a sentinel value to indicate that the parameter was not provided. This is common for distinguishing between "no value provided" and None.
And I've add `type: ignore` to ignore that line.

Pyright and older basedpyright is not raising error here.

**Comment by KotlinIsland** [comment]
this was caused by the same underlying issue as https://github.com/DetachHead/basedpyright/issues/1119#issuecomment-2692490282, and will be fixed in the next release (likely in the next two days)

**Comment by CNSeniorious000** [comment]
Sorry, but it seems this false positive remains in 1.28.1

**Comment by DetachHead** [comment]
i usually close issues as soon as they are fixed in the main branch rather than when i do a release, sorry for the confusion

**Comment by CNSeniorious000** [comment]
My mistake 🙏 I misread the a Dependabot PR and thought 1.28.1 was just released. My apologies. Thank you for the clarification! keep up your own pace!
